test/fake/fake-swap-plugin.ts

```diff
@@ -16,19 +17,26 @@ const swapInfo: EdgeSwapInfo = {
   supportEmail: 'support@fakeswap'
 }
 
+const asFakeSwapSettings = asObject({
+  kycToken: asOptional(asString)
+})
+
 export const fakeSwapPlugin: EdgeSwapPlugin = {
   swapInfo,
 
-  checkSettings(userSettings: JsonObject): EdgeSwapPluginStatus {
-    return { needsActivation: typeof userSettings.kycToken !== 'string' }
+  checkSettings(userSettings: object): EdgeSwapPluginStatus {
+    const cleanSettings = asFakeSwapSettings(userSettings)
+    return { needsActivation: cleanSettings.kycToken == null }
   },
 
   fetchSwapQuote(
     request: EdgeSwapRequest,
-    userSettings: JsonObject = {}
+    userSettings: object = {}
   ): Promise<EdgeSwapQuote> {
+    const cleanSettings = asFakeSwapSettings(userSettings)
+
     // We need KYC:
-    if (typeof userSettings.kycToken !== 'string') {
+    if (typeof cleanSettings.kycToken !== 'string') {
```

Comment:
I'm just a teeny bit annoyed you changed the checkSettings typeof kycToken comparison to a null check but not this one

Reply:
Derp! Good catch.